AdamantTests/Modules/Wallets/DashWalletServiceIntegrationTests.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  DashWalletServiceIntegrationTests.swift
+//  Adamant
+//
+//  Created by Christian Benua on 24.01.2025.
+//  Copyright © 2025 Adamant. All rights reserved.
+//
+
+import XCTest
+@testable import Adamant
+import Swinject
+import BitcoinKit
+import CommonKit
+
+final class DashWalletServiceIntegrationTests: XCTestCase {
+    
+    private var apiCoreMock: APICoreProtocolMock!
+    private var lastTransactionStorageMock: DashLastTransactionStorageProtocolMock!
+    private var dashApiServiceProtocolMock: DashApiServiceProtocolMock!
+    private var sut: DashWalletService!
+    
+    override func setUp() {
+        super.setUp()
+        apiCoreMock = APICoreProtocolMock()
+        dashApiServiceProtocolMock = DashApiServiceProtocolMock()
+        dashApiServiceProtocolMock.api = DashApiCore(apiCore: apiCoreMock)
+        lastTransactionStorageMock = DashLastTransactionStorageProtocolMock()
+        
+        sut = DashWalletService()
+        sut.lastTransactionStorage = lastTransactionStorageMock
+        sut.addressConverter = AddressConverterFactory().make(network: DashMainnet())
+        sut.dashApiService = dashApiServiceProtocolMock
+        sut.transactionFactory = BitcoinKitTransactionFactory()
+    }
+    
+    override func tearDown() {
+        apiCoreMock = nil
+        dashApiServiceProtocolMock = nil
+        lastTransactionStorageMock = nil
+        sut = nil
+        super.tearDown()
+    }
+    
+    func test_createAndSendTransaction_createsValidTxIdAndHash() async throws {
+        // given
+        sut.setWalletForTests(try makeWallet())
+        let data = Constants.unspentTranscationsData
+        await apiCoreMock.isolated { mock in
+            mock.stubbedSendRequestBasicGenericResult = APIResponseModel(result: .success(data), data: data, code: 200)
+        }
+        
+        // when 1
+        let result = await Result(catchingAsync: {
+            try await self.sut.createTransaction(
+                recipient: "Xp6kFbogHMD4QRBDLQdqRp5zUgzmfj1KPn",
+                amount: 0.01,
+                fee: 0.0000001,
+                comment: nil
+            )
+        })
+        
+        // then 1
```

Comment:
What about splitting this UT into 3 separate onces? 1 given/when/expect per test?
It may introduce a bit of code duplication, apart from re-using the same code for configuration put in a separate methods with arguments,
but in the long term it is:

a) easier to code review
b) easier to modify

What do you think @ChrisBenua?

Reply:
In my point of view, integration test supposed to be like that, because we're testing whole flow, from creating transaction to sending it and checking after each action that everything is ok/

Moreover, if I split it, I will have at least 20 lines of the same code in each testing methods (creating transaction). 

I'm not sure its good idea to make two testing methods totally ~55 lines long instead of 35 lines long single test function.